src/ui/contexts/TransactionsContext.tsx

```diff
@@ -37,52 +38,59 @@ export function TransactionsContextProvider({
 
     const { extrinsic, accountId, isValid, onSuccess, onError } = tx;
     setTxs({ ...txs, [id]: { ...tx, status: TxStatusMap.Processing } });
-    const injector = systemChainType.isDevelopment ? undefined : await web3FromAddress(accountId);
-    const account = systemChainType.isDevelopment ? keyring.getPair(accountId) : accountId;
+
+    const keyPair = keyring.getPair(accountId);
+    let addressOrPair: AddressOrPair = keyPair;
+    let signer = undefined;
+
+    // If the account is not a testing account (//Alice etc.),
+    // we need the `signer` from the extension
+    if (!keyPair.meta.isTesting) {
+      signer = (await web3FromAddress(accountId)).signer;
+      // Only use plain address, otherwise pjs api want to sign with given KeyPair
+      // instead of `signer`
+      addressOrPair = keyPair.address;
+    }
 
     try {
-      const unsub = await extrinsic.signAndSend(
-        account,
-        { signer: injector?.signer || undefined },
-        async result => {
-          if (result.isInBlock || result.isFinalized) {
-            const events: Record<string, number> = {};
-
-            result.events.forEach(record => {
-              const { event } = record;
-              const key = `${event.section}:${event.method}`;
-              if (!events[key]) {
-                events[key] = 1;
-              } else {
-                events[key]++;
-              }
-            });
-
-            if (!isValid(result)) {
-              setTxs({ ...txs, [id]: { ...tx, status: TxStatusMap.Error, events } });
-
-              let message = 'Transaction failed';
-
-              if (result.dispatchError?.isModule) {
-                const decoded = api?.registry.findMetaError(result.dispatchError.asModule);
-                message = `${decoded?.section.toUpperCase()}.${decoded?.method}: ${decoded?.docs}`;
-              }
-
-              onError && onError(result);
-
-              throw new Error(message);
+      const unsub = await extrinsic.signAndSend(addressOrPair, { signer }, async result => {
+        if (result.isInBlock || result.isFinalized) {
+          const events: Record<string, number> = {};
+
+          result.events.forEach(record => {
+            const { event } = record;
+            const key = `${event.section}:${event.method}`;
+            if (!events[key]) {
+              events[key] = 1;
+            } else {
+              events[key]++;
```

Comment:
I think we need a conditional check here. Is there a range of allowed values?

Reply:
The code in the `try` block just got reformatted. 
While looking into it, I'm less of a fan how this is done. However, refactoring should be done another day.

The `extrinsic.signAndSend()` call registers a callback, which is called for every state the extrinsic is in. The section you pointed out collects the events from each state. Not in a list but in a map from `event => amount of occurrence`. Weird choice but let's not break things. 🙈